apps/designer/app/shared/context.server.ts

```diff
@@ -0,0 +1,58 @@
+import type { AppContext } from "@webstudio-is/trpc-interface/server";
+import { z } from "zod";
+import * as cryptoJson from "./crypto/crypto-json.server";
+import { authenticator } from "~/services/auth.server";
+import { trpcClient } from "~/services/trpc.server";
+import { getMode } from "./router-utils/build-params";
+
+const ReadToken = z.object({
+  projectId: z.string(),
+});
+
+export type ReadToken = z.infer<typeof ReadToken>;
+
+const createAuthorizationContext = async (
+  request: Request
+): Promise<AppContext["authorization"]> => {
+  const url = new URL(request.url);
+
+  const buildMode = getMode(url);
+
+  const readTokenRaw = url.searchParams.get("readToken") ?? undefined;
+  const token = url.searchParams.get("token") ?? url.hostname;
+
+  const user = await authenticator.isAuthenticated(request);
+
+  let readToken: ReadToken | undefined;
+
+  if (readTokenRaw !== undefined) {
+    readToken = ReadToken.parse(
+      await cryptoJson.decode(
+        readTokenRaw,
+        process.env.AUTH_SECRET ?? "NO-SECRET"
```

Comment:
This needs to be added to .env, .env.example and env-check?

Reply:
Already there, reused existing Auth secret key